clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3717,6 +3717,14 @@ def err_sme_definition_using_za_in_non_sme_target : Error<
   "function using ZA state requires 'sme'">;
 def err_sme_definition_using_zt0_in_non_sme2_target : Error<
   "function using ZT0 state requires 'sme2'">;
+def warn_sme_streaming_pass_return_vl_to_non_streaming : Warning<
+  "passing a VL-dependent argument to/from a function that has a different"
+  " streaming-mode, is undefined behaviour">,
```

Comment:
it is only undefined behaviour if the streaming VL != non-streaming VL.

Reply:
Done.